source/simulator/solver_schemes.cc

```diff
@@ -226,17 +226,20 @@ namespace aspect
   {
     // Advect the particles before they are potentially used to
     // set up the compositional fields.
-    if (particle_world.get() != nullptr)
+    for (unsigned int particle_world_index = 0 ; particle_world_index < this->particle_worlds.size(); ++particle_world_index)
       {
-        // Do not advect the particles in the initial refinement stage
-        const bool in_initial_refinement = (timestep_number == 0)
-                                           && (pre_refinement_step < parameters.initial_adaptive_refinement);
-        if (!in_initial_refinement)
-          // Advance the particles in the world to the current time
-          particle_world->advance_timestep();
-
-        if (particle_world->get_property_manager().need_update() == Particle::Property::update_output_step)
-          particle_world->update_particles();
+        if (particle_worlds[particle_world_index].get() != nullptr)
```

Comment:
If you use the range-based for-loop I suggested above, you do not even need the check for the nullpointer (assuming the vector only ever contains existing particle worlds).